repo transfer

SylvainMartel opened the pull request:
view readme file to see current status.

swcurran commented:
@andrewwhitehead , @AlexanderShenshin — can you please review this to make sure @SylvainMartel hasn’t put anything malicious in this.  I’m sure he hasn’t but someone that can read Rust should be the one that confirms that :-). 

Your review may not be sufficient to merge, but I can approve thereafter.

Thanks!

SylvainMartel commented:
I was using the readme file to track progress, but I forgot I emptied it when I cleaned the code to remove debugging and non-working code.  So I added an AI generated status file to the PR to quickly show where the code stands at currently compared to the specs.  I didn't review it extensively, but a quick overview seems like it's correct.  I was about to redo once more the SCID generation, hash verification(it's where I always get stuck, something is wrong in my logic, or the way I understand the specs.)

PatStLouis commented:
I gave the cli a try and was able to create / resolve a did.

@stormer78  Is the `spec_1_0_pre` files needed moving forward? Seems confusing to include this and referring to it as version0.9.

Only minor issue I have is that even if there's no witnesses involved, the resolver will raise an error if the witness file doesn't exist. Otherwise this looks good to me!

stormer78 commented:
> I gave the cli a try and was able to create / resolve a did.
> 
> @stormer78 Is the `spec_1_0_pre` files needed moving forward? Seems confusing to include this and referring to it as version0.9.
> 
> Only minor issue I have is that even if there's no witnesses involved, the resolver will raise an error if the witness file doesn't exist. Otherwise this looks good to me!

As Rust is strongly typed, and the version 1.0 specification has two different possible state types, I needed to have a struct that represents the differences (it is only used to read the old v1.0 spec where null's could be present). So the spec_1_0_pre only exists to correctly read in and convert between the two different specificiations for 1.0 (nulls --> empty lists/objects)

Will address the witness file missing error in next version
